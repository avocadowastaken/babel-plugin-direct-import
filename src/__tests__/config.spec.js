const { prepareConfig } = require("../config");

describe("config", () => {
  describe("prepareConfig", () => {
    it("should throw when for unknown keys", () => {
      expect(() =>
        prepareConfig({ foo: "bar", baz: 10 })
      ).toThrowErrorMatchingSnapshot();
    });

    it("should not throw when given only a name as string", () => {
      expect(() => prepareConfig("lodash")).not.toThrow();
    });

    it("should throw when `name` is not a string", () => {
      expect(() => prepareConfig({ name: 10 })).toThrowErrorMatchingSnapshot();
      expect(() => prepareConfig({ name: {} })).toThrowErrorMatchingSnapshot();
      expect(() => prepareConfig({ name: [] })).toThrowErrorMatchingSnapshot();
      expect(() =>
        prepareConfig({ name: null })
      ).toThrowErrorMatchingSnapshot();
    });

    it("should throw when `name` is an empty string", () => {
      expect(() => prepareConfig({ name: "" })).toThrowErrorMatchingSnapshot();
    });

    it("should throw when `indexFile` is not a string", () => {
      expect(() =>
        prepareConfig({ name: "lodash", indexFile: 10 })
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        prepareConfig({ name: "lodash", indexFile: {} })
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        prepareConfig({ name: "lodash", indexFile: [] })
      ).toThrowErrorMatchingSnapshot();
    });

    it("should not throw when `indexFile` is null or undefined", () => {
      expect(() =>
        prepareConfig({ name: "lodash", indexFile: null })
      ).not.toThrow();
      expect(() => prepareConfig({ name: "lodash" })).not.toThrow();
    });

    it("should throw when `indexFile` is an empty string", () => {
      expect(() =>
        prepareConfig({ name: "lodash", indexFile: "" })
      ).toThrowErrorMatchingSnapshot();
    });

    it("should throw when `indexFile` does not belongs to package", () => {
      expect(() =>
        prepareConfig({ name: "lodash", indexFile: "lodash-es/index" })
      ).toThrowErrorMatchingSnapshot();
    });

    it("should throw when `indexFileContent` is not a string", () => {
      expect(() =>
        prepareConfig({ name: "lodash", indexFileContent: 10 })
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        prepareConfig({ name: "lodash", indexFileContent: {} })
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        prepareConfig({ name: "lodash", indexFileContent: [] })
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        prepareConfig({ name: "lodash", indexFileContent: null })
      ).toThrowErrorMatchingSnapshot();
    });

    it("should throw when `indexFileContent` is an empty string", () => {
      expect(() =>
        prepareConfig({ name: "lodash", indexFileContent: "" })
      ).toThrowErrorMatchingSnapshot();
    });

    it("should always return array", () => {
      const config = prepareConfig({ name: "lodash" });

      expect(Array.isArray(config)).toBe(true);
    });

    it("should create normalized config", () => {
      expect(
        prepareConfig({ name: "lodash", indexFile: "lodash/index" })
      ).toEqual([
        { name: "lodash", indexFile: "lodash/index", indexFileContent: null },
      ]);

      expect(prepareConfig("material-ui")).toEqual([
        { name: "material-ui", indexFile: null, indexFileContent: null },
      ]);
    });
  });
});
