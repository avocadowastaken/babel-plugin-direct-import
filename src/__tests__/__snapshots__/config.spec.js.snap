// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config prepareConfig should throw when \`indexFile\` does not belongs to package 1`] = `"babel-plugin-direct-import: Index file \\"lodash-es/index\\" must belong to \\"lodash\\" package"`;

exports[`config prepareConfig should throw when \`indexFile\` is an empty string 1`] = `"babel-plugin-direct-import: { indexFile } is empty"`;

exports[`config prepareConfig should throw when \`indexFile\` is not a string 1`] = `"babel-plugin-direct-import: { indexFile } expected to be a string"`;

exports[`config prepareConfig should throw when \`indexFile\` is not a string 2`] = `"babel-plugin-direct-import: { indexFile } expected to be a string"`;

exports[`config prepareConfig should throw when \`indexFile\` is not a string 3`] = `"babel-plugin-direct-import: { indexFile } expected to be a string"`;

exports[`config prepareConfig should throw when \`indexFile\` is not a string 4`] = `"babel-plugin-direct-import: { indexFile } expected to be a string"`;

exports[`config prepareConfig should throw when \`indexFileContent\` is an empty string 1`] = `"babel-plugin-direct-import: { indexFileContent } is empty"`;

exports[`config prepareConfig should throw when \`indexFileContent\` is not a string 1`] = `"babel-plugin-direct-import: { indexFileContent } expected to be a string"`;

exports[`config prepareConfig should throw when \`indexFileContent\` is not a string 2`] = `"babel-plugin-direct-import: { indexFileContent } expected to be a string"`;

exports[`config prepareConfig should throw when \`indexFileContent\` is not a string 3`] = `"babel-plugin-direct-import: { indexFileContent } expected to be a string"`;

exports[`config prepareConfig should throw when \`indexFileContent\` is not a string 4`] = `"babel-plugin-direct-import: { indexFileContent } expected to be a string"`;

exports[`config prepareConfig should throw when \`name\` is an empty string 1`] = `"babel-plugin-direct-import: { name } is empty"`;

exports[`config prepareConfig should throw when \`name\` is not a string 1`] = `"babel-plugin-direct-import: { name } expected to be a string"`;

exports[`config prepareConfig should throw when \`name\` is not a string 2`] = `"babel-plugin-direct-import: { name } expected to be a string"`;

exports[`config prepareConfig should throw when \`name\` is not a string 3`] = `"babel-plugin-direct-import: { name } expected to be a string"`;

exports[`config prepareConfig should throw when \`name\` is not a string 4`] = `"babel-plugin-direct-import: { name } expected to be a string"`;

exports[`config prepareConfig should throw when for unknown keys 1`] = `"babel-plugin-direct-import: contains unknown keys { foo, baz }"`;
